[BUG] window_function_test.py::test_window_ride_along failed in databricks runtime

**Describe the bug**
tests added in https://github.com/NVIDIA/spark-rapids/pull/3117, could be missing db-shims updates

**pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class com.databricks.sql.execution.window.RunningWindowFunctionExec**

failures list,
```
[2021-08-05T05:42:14.097Z] FAILED ../../src/main/python/window_function_test.py::test_window_ride_along[Map(String(not_null),String)][IGNORE_ORDER({'local': True})]
[2021-08-05T05:42:14.097Z] FAILED ../../src/main/python/window_function_test.py::test_window_ride_along[Map(RepeatSeq(not_null)(Integer(not_null)),Long)][IGNORE_ORDER({'local': True})]
[2021-08-05T05:42:14.097Z] FAILED ../../src/main/python/window_function_test.py::test_window_ride_along[Map(Boolean(not_null),Boolean)][IGNORE_ORDER({'local': True})]
[2021-08-05T05:42:14.097Z] FAILED ../../src/main/python/window_function_test.py::test_window_ride_along[Map(String(not_null),Array(String))][IGNORE_ORDER({'local': True})]
[2021-08-05T05:42:14.097Z] FAILED ../../src/main/python/window_function_test.py::test_window_ride_along[Map(String(not_null),Map(String(not_null),String))][IGNORE_ORDER({'local': True})]
```

detailed log,
```
[2021-08-05T05:42:13.426Z] =================================== FAILURES ===================================
[2021-08-05T05:42:13.426Z] [31m[1m_____________ test_window_ride_along[Map(String(not_null),String)] _____________[0m
[2021-08-05T05:42:13.426Z] [gw3] linux -- Python 3.8.8 /databricks/conda/envs/databricks-ml-gpu/bin/python
[2021-08-05T05:42:13.426Z] 
[2021-08-05T05:42:13.426Z] ride_along = Map(String(not_null),String)
[2021-08-05T05:42:13.426Z] 
[2021-08-05T05:42:13.426Z]     [37m@ignore_order[39;49;00m(local=[94mTrue[39;49;00m)
[2021-08-05T05:42:13.426Z]     [37m@pytest[39;49;00m.mark.parametrize([33m'[39;49;00m[33mride_along[39;49;00m[33m'[39;49;00m, all_basic_gens + decimal_gens + array_gens_sample + struct_gens_sample + map_gens_sample, ids=idfn)
[2021-08-05T05:42:13.426Z]     [94mdef[39;49;00m [92mtest_window_ride_along[39;49;00m(ride_along):
[2021-08-05T05:42:13.426Z] >       assert_gpu_and_cpu_are_equal_sql(
[2021-08-05T05:42:13.426Z]                 [94mlambda[39;49;00m spark : gen_df(spark, [([33m'[39;49;00m[33ma[39;49;00m[33m'[39;49;00m, LongRangeGen()), ([33m'[39;49;00m[33mb[39;49;00m[33m'[39;49;00m, ride_along)]),
[2021-08-05T05:42:13.426Z]                 [33m"[39;49;00m[33mwindow_agg_table[39;49;00m[33m"[39;49;00m,
[2021-08-05T05:42:13.426Z]                 [33m'[39;49;00m[33mselect *,[39;49;00m[33m'[39;49;00m
[2021-08-05T05:42:13.426Z]                 [33m'[39;49;00m[33m row_number() over (order by a) as row_num [39;49;00m[33m'[39;49;00m
[2021-08-05T05:42:13.426Z]                 [33m'[39;49;00m[33mfrom window_agg_table [39;49;00m[33m'[39;49;00m,
[2021-08-05T05:42:13.426Z]                 conf = allow_negative_scale_of_decimal_conf)
[2021-08-05T05:42:13.426Z] 
[2021-08-05T05:42:13.426Z] [1m[31m../../src/main/python/window_function_test.py[0m:846: 
[2021-08-05T05:42:13.426Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2021-08-05T05:42:13.426Z] [1m[31m../../src/main/python/asserts.py[0m:476: in assert_gpu_and_cpu_are_equal_sql
[2021-08-05T05:42:13.426Z]     assert_gpu_and_cpu_are_equal_collect(do_it_all, conf, is_cpu_first=is_cpu_first)
[2021-08-05T05:42:13.426Z] [1m[31m../../src/main/python/asserts.py[0m:434: in assert_gpu_and_cpu_are_equal_collect
[2021-08-05T05:42:13.426Z]     _assert_gpu_and_cpu_are_equal(func, [33m'[39;49;00m[33mCOLLECT[39;49;00m[33m'[39;49;00m, conf=conf, is_cpu_first=is_cpu_first)
[2021-08-05T05:42:13.426Z] [1m[31m../../src/main/python/asserts.py[0m:415: in _assert_gpu_and_cpu_are_equal
[2021-08-05T05:42:13.426Z]     run_on_gpu()
[2021-08-05T05:42:13.426Z] [1m[31m../../src/main/python/asserts.py[0m:409: in run_on_gpu
[2021-08-05T05:42:13.426Z]     from_gpu = with_gpu_session(bring_back, conf=conf)
[2021-08-05T05:42:13.426Z] [1m[31m../../src/main/python/spark_session.py[0m:105: in with_gpu_session
[2021-08-05T05:42:13.426Z]     [94mreturn[39;49;00m with_spark_session(func, conf=copy)
[2021-08-05T05:42:13.426Z] [1m[31m../../src/main/python/spark_session.py[0m:70: in with_spark_session
[2021-08-05T05:42:13.426Z]     ret = func(_spark)
[2021-08-05T05:42:13.426Z] [1m[31m../../src/main/python/asserts.py[0m:190: in <lambda>
[2021-08-05T05:42:13.426Z]     bring_back = [94mlambda[39;49;00m spark: limit_func(spark).collect()
[2021-08-05T05:42:13.426Z] [1m[31m/databricks/spark/python/pyspark/sql/dataframe.py[0m:697: in collect
[2021-08-05T05:42:13.426Z]     sock_info = [96mself[39;49;00m._jdf.collectToPython()
[2021-08-05T05:42:13.426Z] [1m[31m/databricks/spark/python/lib/py4j-0.10.9-src.zip/py4j/java_gateway.py[0m:1304: in __call__
[2021-08-05T05:42:13.426Z]     return_value = get_return_value(
[2021-08-05T05:42:13.426Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2021-08-05T05:42:13.426Z] 
[2021-08-05T05:42:13.426Z] a = ('xro501424', <py4j.java_gateway.GatewayClient object at 0x7fac1675e340>, 'o501421', 'collectToPython')
[2021-08-05T05:42:13.427Z] kw = {}
[2021-08-05T05:42:13.427Z] converted = IllegalArgumentException('Part of the plan is not columnar class com.databricks.sql.execution.window.RunningWindowFunc...:79)\n\tat py4j.GatewayConnection.run(GatewayConnection.java:251)\n\tat java.lang.Thread.run(Thread.java:748)\n', None)
[2021-08-05T05:42:13.427Z] 
[2021-08-05T05:42:13.427Z]     [94mdef[39;49;00m [92mdeco[39;49;00m(*a, **kw):
[2021-08-05T05:42:13.427Z]         [94mtry[39;49;00m:
[2021-08-05T05:42:13.427Z]             [94mreturn[39;49;00m f(*a, **kw)
[2021-08-05T05:42:13.427Z]         [94mexcept[39;49;00m py4j.protocol.Py4JJavaError [94mas[39;49;00m e:
[2021-08-05T05:42:13.427Z]             converted = convert_exception(e.java_exception)
[2021-08-05T05:42:13.427Z]             [94mif[39;49;00m [95mnot[39;49;00m [96misinstance[39;49;00m(converted, UnknownException):
[2021-08-05T05:42:13.427Z]                 [90m# Hide where the exception came from that shows a non-Pythonic[39;49;00m
[2021-08-05T05:42:13.427Z]                 [90m# JVM exception message.[39;49;00m
[2021-08-05T05:42:13.427Z] >               [94mraise[39;49;00m converted [94mfrom[39;49;00m [96mNone[39;49;00m
[2021-08-05T05:42:13.427Z] [1m[31mE               pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class com.databricks.sql.execution.window.RunningWindowFunctionExec[0m
[2021-08-05T05:42:13.427Z] [1m[31mE               RunningWindowFunction [a#159921L, b#159922, row_number() windowspecdefinition(a#159921L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS row_num#159927], [a#159921L ASC NULLS FIRST], false[0m
[2021-08-05T05:42:13.427Z] [1m[31mE               +- GpuColumnarToRow false[0m
[2021-08-05T05:42:13.427Z] [1m[31mE                  +- GpuSort [a#159921L ASC NULLS FIRST], false, com.nvidia.spark.rapids.OutOfCoreSort$@85a7c51[0m
[2021-08-05T05:42:13.427Z] [1m[31mE                     +- GpuShuffleCoalesce 2147483647[0m
[2021-08-05T05:42:13.427Z] [1m[31mE                        +- GpuColumnarExchange gpusinglepartitioning$(), ENSURE_REQUIREMENTS, [id=#247289][0m
[2021-08-05T05:42:13.427Z] [1m[31mE                           +- GpuRowToColumnar targetsize(2147483647)[0m
[2021-08-05T05:42:13.427Z] [1m[31mE                              +- Scan ExistingRDD[a#159921L,b#159922][0m
[2021-08-05T05:42:13.427Z] 
```

--- a/shims/spark301db/src/main/scala/com/nvidia/spark/rapids/shims/spark301db/Spark301dbShims.scala
+++ b/shims/spark301db/src/main/scala/com/nvidia/spark/rapids/shims/spark301db/Spark301dbShims.scala
@@ -121,7 +121,8 @@ class Spark301dbShims extends Spark301Shims {
             "i.e. (UNBOUNDED PRECEDING TO CURRENT ROW)",
         ExecChecks(
           (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL +
-              TypeSig.STRUCT + TypeSig.ARRAY).nested() +
+              TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP).nested() +
+              TypeSig.psNote(TypeEnum.MAP, "Not supported as a partition by key") +
               TypeSig.psNote(TypeEnum.STRUCT, "Not supported as a partition by key") +
               TypeSig.psNote(TypeEnum.ARRAY, "Not supported as a partition by key"),
           TypeSig.all),

--- a/shims/spark311db/src/main/scala/com/nvidia/spark/rapids/shims/spark311db/Spark311dbShims.scala
+++ b/shims/spark311db/src/main/scala/com/nvidia/spark/rapids/shims/spark311db/Spark311dbShims.scala
@@ -109,7 +109,8 @@ class Spark311dbShims extends Spark311Shims {
             "i.e. (UNBOUNDED PRECEDING TO CURRENT ROW)",
         ExecChecks(
           (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL +
-              TypeSig.STRUCT + TypeSig.ARRAY).nested() +
+              TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP).nested() +
+              TypeSig.psNote(TypeEnum.MAP, "Not supported as a partition by key") +
               TypeSig.psNote(TypeEnum.STRUCT, "Not supported as a partition by key") +
               TypeSig.psNote(TypeEnum.ARRAY, "Not supported as a partition by key"),
             TypeSig.all),

